exports/dataset.cpp

```diff
@@ -1084,12 +1147,19 @@ PYBIND11_MODULE(dataset, m) {
              return self(key.first, key.second);
            },
            py::keep_alive<0, 1>())
-      .def("copy", [](const Dataset &self) { return self; })
+      .def("copy", [](const Dataset &self) { return self; },
+           "Make a copy of a Dataset.")
       .def("__copy__", [](Dataset &self) { return Dataset(self); })
       .def("__deepcopy__",
            [](Dataset &self, py::dict) { return Dataset(self); })
-      .def_property_readonly("subset",
-                             [](Dataset &self) { return SubsetHelper(self); })
+      .def_property_readonly(
+          "subset", [](Dataset &self) { return SubsetHelper(self); },
+          "Used to extract a read-only subset of the Dataset in two different "
+          "ways: \n - one can use just a string, e.g. d.subset['sample'] to "
+          "extract the sample and its variance, and all the relevant "
+          "coordinates.\n - one can also use a tag/string combination for a "
+          "more restrictive extraction, e.g. d.subset[Data.Value, 'sample'] to "
+          "only get the Value and associated coordinates.")
```

Comment:
Same as above. Also, you get a slice back not a new `Dataset` don't know if you think that is worth mentioning?